scripts/ci/jobs/diff-dumps.sh

```diff
@@ -107,70 +107,87 @@ create_offline_dump() {
-    while read -r version; do
-        status_code=$(curl --silent --max-time 60 --retry 3 --write-out "%{http_code}" -o /dev/null "https://github.com/stackrox/stackrox/releases/tag/$version.0")
-        if [[ "$status_code" == "404" ]]; then
-            # Release was not cut, so try a previous release.
-            continue
-        elif [[ "$status_code" == "200" ]]; then
-            # Release was cut, so use this one.
-            latest_version="$version"
-            break
-        else
-            # Some other error occurred.
-            >&2 echo "received status code $status_code when fetching StackRox releases"
-            exit 1
-        fi
-    done < <(grep -oE '^[0-9]+\.[0-9]+' "$version_file" | sort --reverse --unique --version-sort)
-    if [[ -z "$latest_version" ]]; then
-      >&2 echo "programmer error: latest_version unset..."
-      exit 1
-    fi
-
-    file_to_check="scanner-v4-defs-${latest_version}.zip"
-
-    curl --silent --show-error --fail --max-time 60 --retry 3 -o "$file_to_check" "https://definitions.stackrox.io/v4/offline-bundles/$file_to_check"
-    unzip -l "$file_to_check"
-    zip scanner-vuln-updates.zip "$file_to_check"
-    echo "$file_to_check added to scanner-vuln-updates.zip"
-
-    cmd=()
-    if is_in_PR_context; then
-        cmd+=(echo "Would do")
-    fi
-    "${cmd[@]}" gsutil cp scanner-vuln-updates.zip gs://scanner-support-public/offline/v1/scanner-vuln-updates.zip
+# scanner_v4_create_and_add_bundles() creates versioned offline bundles for V4
+# and add V4 definitions to the latest offline bundle.  It determines the
+# version to use for each release based on the contents of the
+# VULNERABILITY_VERSION file present in each release tag.
+#
+scanner_v4_create_and_add_bundles() {
+    local latest_bundle="${1:-scanner-vuln-updates.zip}"
+
+    local curl="curl --silent --show-error --max-time 60 --retry 3"
+    local release_version_url="https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/version/RELEASE_VERSION"
+    local vuln_version_pattern="https://raw.githubusercontent.com/stackrox/stackrox/%s/scanner/VULNERABILITY_VERSION"
+
+    [ -f scanner-defs.zip ] || die "missing required bundle inclusion: scanner-defs.zip"
+    [ -f k8s-istio.zip ]    || die "missing required bundle inclusion: k8s-istio.zip"
+    [ -f "$latest_bundle" ] || die "missing latest bundle: $latest_bundle"
+
+    info "existing latest bundle:"
+    unzip -l "$latest_bundle"
+
+    info "fetching release versions at $release_version_url"
+
+    # Read the supported releases file, filter lines with release versions,
+    # parse 4.4.* and 4.5.* to print only the X.Y, otherwise print the whole
+    # X.Y.Z, finally merge everything together, and pipe one release per line.
+
+    $curl --fail "$release_version_url" \
+        | grep '^[0-9]\+\.[0-9]\+\.[0-9]\+$' \
+        | sed 's/^\(4\.4\|4\.5\)\..*/\1/' \
+        | sort -V \
+        | uniq \
```

Comment:
I made a file as such:

```
4.4.0
4.4.1
4.4.2
4.4.3
4.4.4
4.4.5
4.5.0
4.5.1
4.5.2
4.5.3
4.5.4-rc.1
4.6.0
```

and I ran it through all commands until after `uniq`, and it looks the same as before

Reply:
`4.6.0` should be there.

```
⮩  cat RELEASE_VERSION
# This file serves as a list of StackRox versions.
# As Scanner V4 and StackRox versions are aligned, the vulnerability updaters are responsible for updating vulnerability data across all versions listed in this file.
# Subsequently, this updated data is uploaded to Google Cloud Storage.
# For each individual version of Scanner V4's vulnerability updater, corresponding vulnerability data is collected and then seamlessly uploaded to Google Cloud Storage.
# This process ensures that each version's vulnerability data is accurately captured and maintained in accordance with its respective version.
4.4.0
4.4.1
4.4.2
4.4.3
4.4.4
4.4.5
4.5.0
4.5.1
4.5.2
4.5.3
4.6.0
4.5.4-rc.1
⮩  cat RELEASE_VERSION | grep 4.6.0
4.6.0
⮩  cat RELEASE_VERSION | grep '^[0-9]\+\.[0-9]\+\.[0-9]\+$' | sed 's/^\(4\.4\|4\.5\)\..*/\1/'
4.4
4.4
4.4
4.4
4.4
4.4
4.5
4.5
4.5
4.5
4.6.0
⮩  cat RELEASE_VERSION | grep '^[0-9]\+\.[0-9]\+\.[0-9]\+$' | sed 's/^\(4\.4\|4\.5\)\..*/\1/'  | grep 4.6.0
4.6.0
⮩  
```

Reply:
```
$ cat scanner/updater/version/RELEASE_VERSION
# This file serves as a list of StackRox versions.
# As Scanner V4 and StackRox versions are aligned, the vulnerability updaters are responsible for updating vulnerability data across all versions listed in this file.
# Subsequently, this updated data is uploaded to Google Cloud Storage.
# For each individual version of Scanner V4's vulnerability updater, corresponding vulnerability data is collected and then seamlessly uploaded to Google Cloud Storage.
# This process ensures that each version's vulnerability data is accurately captured and maintained in accordance with its respective version.
4.4.0
4.4.1
4.4.2
4.4.3
4.4.4
4.4.5
4.5.0
4.5.1
4.5.2
4.5.3
4.6.0
4.5.4-rc.1
$ cat scanner/updater/version/RELEASE_VERSION |  grep '^[0-9]\+\.[0-9]\+\.[0-9]\+$' | sed 's/^\(4\.4\|4\.5\)\..*/\1/'  | sort -V | uniq
4.4.0
4.4.1
4.4.2
4.4.3
4.4.4
4.4.5
4.5.0
4.5.1
4.5.2
4.5.3
4.6.0
```

There must be something wrong with my `sed` (builtin `sed` on macOS)

I tried again in a UBI 8 image:

```
# cat RELEASE_VERSION
# This file serves as a list of StackRox versions.
# As Scanner V4 and StackRox versions are aligned, the vulnerability updaters are responsible for updating vulnerability data across all versions listed in this file.
# Subsequently, this updated data is uploaded to Google Cloud Storage.
# For each individual version of Scanner V4's vulnerability updater, corresponding vulnerability data is collected and then seamlessly uploaded to Google Cloud Storage.
# This process ensures that each version's vulnerability data is accurately captured and maintained in accordance with its respective version.
4.4.0
4.4.1
4.4.2
4.4.3
4.4.4
4.4.5
4.5.0
4.5.1
4.5.2
4.5.3
4.6.0
4.5.4-rc.1
# cat RELEASE_VERSION | grep '^[0-9]\+\.[0-9]\+\.[0-9]\+$' | sed 's/^\(4\.4\|4\.5\)\..*/\1/'  | sort -V | uniq
4.4
4.5
4.6.0
```

Reply:
marking this as resolved since we run this on Linux which has a version of `sed` that isn't messed up

Reply:
If you want to run the same `sed` used on GNU/Linux you might want to start with:

```
brew install gnu-sed
```

I've been told that this will install the GNU port of `sed`; I don't think we are using any GNU extension in the sed expression above, but at least the POSIX implementation should be the same.